src/preamble.js

```diff
@@ -1067,6 +1067,12 @@ function createWasm() {
 #endif
 #endif // WASM == 2
 
+#if ASSERTIONS
+      // Warn on some common problems.
+      if (isFileURI(wasmBinaryFile)) {
+        err(`warning: Loading from a file URI (${wasmBinaryFile}) is not supported in most browsers.`);
```

Comment:
Will the show up in the default emscripten html output?  (Not just in the js console?)

Reply:
This probably should be more that just a warning, its completely fatal isn't it?

Reply:
It does not show up on the main HTML, since it goes to stderr. But in the dev console it should still be helpful I think?

This is just a warning because it's already fatal here anyhow. This is just extra logging of a hint, before we die. And note that it might be incorrect, as some environments *might* allow file URIs, like old browsers, and maybe other things.

Reply:
Is it possible though that some folks don't know to open the JS console to see the error message.  Or do we consider that a minimum expectation?

Reply:
That is a risk, yes, but I think we assume it. When a website doesn't work, the first thing a dev should check is the JS console. Someone not familiar with web development might not know, however, so I agree with you. We may want, separately from this PR, to show the stderr output to stdout, I suppose, but I'm not sure how to do that in a way that doesn't have downsides.